pkg/clean/clean.go

```diff
@@ -161,12 +162,18 @@ func InitializeCleanupService(cf *config.NSXOperatorConfig, nsxClient *nsx.Clien
 			return subnetport.InitializeSubnetPort(service)
 		}
 	}
+	wrapInitializeIPAddressAllocation := func(service common.Service) cleanupFunc {
+		return func() (cleanup, error) {
+			return ipaddressallocation.InitializeIPAddressAllocation(service, vpcService)
+		}
+	}
 	// TODO: initialize other CR services
 	cleanupService = cleanupService.
 		AddCleanupService(wrapInitializeSubnetPort(commonService)).
 		AddCleanupService(wrapInitializeSubnetService(commonService)).
 		AddCleanupService(wrapInitializeSecurityPolicy(commonService)).
 		AddCleanupService(wrapInitializeStaticRoute(commonService)).
+		AddCleanupService(wrapInitializeIPAddressAllocation(commonService)).
```

Comment:
I just discussed with Danting that we had a requirement to cleanup the IPAddressAllocations created for the LB. We decided to put that cleanup logic into this IPAddressAllocation service. So could you help change the https://github.com/vmware-tanzu/nsx-operator/blob/35aa0b8fec20bd77ab6fac9eaa2462cf316a5f38/pkg/nsx/services/ipaddressallocation/ipaddressallocation.go#L170-L173 from `ListIndexFuncValues` to `List`?

Reply:
ok